nixos/modules/services/mail/zeyple.nix

```diff
@@ -0,0 +1,91 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+let
+  cfg = config.services.zeyple;
+  bool2int = x: if x then "1" else "0";
+
+  gpgHome = pkgs.runCommand "zeyple-gpg-home" { } ''
+    mkdir -p $out
+    for file in ${lib.concatStringsSep " " cfg.keys}; do
+      ${pkgs.gnupg}/bin/gpg --homedir="$out" --import "$file"
+    done
+
+    # Remove socket files
+    rm -f $out/S.gpg-agent*
+  '';
+in {
+  options.services.zeyple = {
+    enable = mkEnableOption "Zeyple, an utility program to automatically encrypt outgoing emails with GPG";
+
+    user = mkOption {
+      type = types.str;
+      default = "zeyple";
+      description = "User to run Zeyple as.";
+    };
+
+    group = mkOption {
+      type = types.str;
+      default = "zeyple";
+      description = "Group to use to run Zeyple.";
+    };
+
+    logFile = mkOption {
+      type = types.path;
+      default = "/var/log/zeyple.log";
+      description = "Path of the log file.";
+    };
+
+    forceEncrypt = mkOption {
+      type = types.bool;
+      default = true;
+      description = "Whether to abort sending mail if no keys are found.";
+    };
+
+    keys = mkOption {
+      type = with types; listOf path;
+      description = "List of public key files that will be imported by gpg.";
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    users.groups."${cfg.group}" = { };
+    users.users."${cfg.user}" = {
+      isSystemUser = true;
+      group = "${cfg.group}";
+    };
+
+    systemd.tmpfiles.rules = [ "f '${cfg.logFile}' 0600 ${cfg.user} ${cfg.group} - -" ];
```

Comment:
Will this file grow indefinitely? See [services.logrotate](https://search.nixos.org/options?channel=unstable&from=0&size=50&sort=relevance&type=packages&query=services.logrotate.).

NOTE: the `logrotate` service is going to have a few [upcoming changes](https://github.com/NixOS/nixpkgs/pull/162063)... but you can ignore that for now.